packages/redux-store/src/recordListener.ts

```diff
@@ -21,7 +21,20 @@ export const removeRecord = createAction<BaseRecord>('record/remove')
  */
 export const startRecordListeners = (agent: Agent, store: Store) => {
   const onDeleted = (event: RecordDeletedEvent<BaseRecord>) => {
-    store.dispatch(removeRecord(event.payload.record))
+    // Extract the record from event
+    const record = event.payload.record
+    //  Delete the record by id if it is of the generic type
+    const retrieveAndDeleteGenericRecord = async (record: { id: string; type: string }) => {
+      //  No need to fetch it first if we have the id?!
+      await agent.genericRecords.deleteById(record.id)
+    }
+    if (record instanceof BaseRecord) {
+      // BaseRecord
+      store.dispatch(removeRecord(record))
+    } else {
+      // Gen record type
+      void retrieveAndDeleteGenericRecord(record)
+    }
```

Comment:
This code can stay as is. The deleting is already done. We just need to remove it from the store. As the type of `record` could now also be a simple object with keys `id` and `type`, we jsut need to update the removeRecord method to also support the changes made in AFJ. See my other comment on suggestions on how to resolve

```suggestion
    store.dispatch(removeRecord(event.payload.record))
```